server/view.go

```diff
@@ -95,6 +101,138 @@ func serveView(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func serveBatView(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"error":"invalid URL query"}`))
+		return
+	}
+
+	GameID, err := strconv.Atoi(r.Form.Get("id"))
+
+	if err != nil || GameID <= 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"error":"invalid id value"}`))
+		return
+	}
+
+	recordingPosition := getRecordingPositionByGameId(strconv.Itoa(GameID))
+
+	if len(sortedRecordings) == recordingPosition {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"error":"game not found"}`))
+		return
+	}
+
+	recRenderArg := getRecordingArg(recordingPosition, r)
+
+	w.Header().Set("Content-Description", "File Transfer");
+	w.Header().Set("Content-Disposition", "attachment; filename=LOL_Observer_" + recRenderArg.GameID + "_spectate.bat");
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+
+	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+		w.Header().Set("Content-Encoding", "gzip")
+		w.WriteHeader(http.StatusOK)
+
+		writer := gzip.NewWriter(w)
+		if err := pageBatTemplate.Execute(writer, recRenderArg); err != nil {
+			writer.Write([]byte("template render error"))
+			log.Println("render: template error:", err)
+		}
+		writer.Close()
+	} else {
+		w.WriteHeader(http.StatusOK)
+		if err := pageBatTemplate.Execute(w, recRenderArg); err != nil {
+			w.Write([]byte("template render error"))
+			log.Println("render: template error:", err)
+		}
+	}
+}
+
+func getRecordingPositionByGameId(GameID string) int {
+	return sort.Search(len(sortedRecordings), func (i int) bool {
+		rec := sortedRecordings[i]
+		info := rec.rec.RetrieveGameInfo()
+
+        return GameID == info.GameID
+	})
+}
+
+func getRecordingArg(i int, r *http.Request) recordingArg {
```

Comment:
Ideally redefine this function to be `func getRecordingArg(rec *internalRecording, host string) recordingArg`